lib/config/options.rb

```diff
@@ -83,17 +83,7 @@ def reload_from_files(*files)
     end
 
     def to_hash
-      result = {}
-      marshal_dump.each do |k, v|
-        if v.instance_of? Config::Options
-          result[k] = v.to_hash
-        elsif v.instance_of? Array
-          result[k] = descend_array(v)
-        else
-          result[k] = v
-        end
-      end
-      result
+      __convert_to_hash(marshal_dump)
```

Comment:
Well, this change brings no value and only makes the code more complicated. so yes, I would like it to be gone :) By the way, are you running this code in Rails? I have a suspicion it might be connected with #120